File: Swift知识点/15-block 回顾/15-block 回顾/XFHttpTool.m
//
//  XFHttpTool.m
//  15-block 回顾
//
//  Created by xiaofans on 16/8/4.
//  Copyright © 2016年 xiaofan. All rights reserved.
//

#import "XFHttpTool.h"


@interface XFHttpTool ()

@property (nonatomic, copy) void(^callBack)(NSString *name);

@end


@implementation XFHttpTool

- (void)loadData:(void (^)(NSString *))callBack {
    self.callBack = callBack;
    dispatch_async(dispatch_get_global_queue(0, 0), ^{
        NSLog(@"子线程下载数据:%@", [NSThread currentThread]);
        dispatch_async(dispatch_get_main_queue(), ^{
            NSLog(@"拿到数据,进行回调%@", [NSThread currentThread]);
            
            callBack(@"返回的json 数据");
        });
    });
}

@end

File: Swift知识点/16-闭包使用/16-闭包使用/XFHttpTool.swift
//
//  XFHttpTool.swift
//  16-闭包使用
//
//  Created by xiaofans on 16/8/4.
//  Copyright © 2016年 xiaofan. All rights reserved.
//

import UIKit

class XFHttpTool: NSObject {
    // 闭包属性
    var callBack : ((jsonData : String) -> ())?
    
    // 闭包类型  (参数列表) -> (返回值类型)
    func loadData(callBack : (jsonData : String) -> ()) {
        
        self.callBack = callBack
        
        dispatch_async(dispatch_get_global_queue(0, 0)) { () -> Void in
            print("发送网络请求:\(NSThread.currentThread())")
            
            dispatch_async(dispatch_get_main_queue(), { () -> Void in
                print("获得数据, 并进行回调:\(NSThread.currentThread())")
                
                callBack(jsonData: "jsonData 数据")
            })
        }
    }

}
